### ISSUE TITLE
Automate Blocking of PR Merge before QA Approval

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va-mobile-app#6261]
This should be a Github actions workflow that:

1. Runs on each PR Open
2. Runs on each PR "approval"
3. Only runs when merges are going into `develop`
4. Only runs on code that requires both FE and QA review

**Also**:

- It should run immediately with a failure, blocking merged for code QA needs to review
- On each approval, it should rerun to test that there are at least 3 approvals, one of which is from a member of the QA team ([flagship-mobile-qa](https://github.com/orgs/department-of-veterans-affairs/teams/flagship-mobile-qa/members)). Ideally, if we can check team membership that would be best, but if not having users listed directly is fine.
- the [CODEOWNERS](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/.github/CODEOWNERS) file outlines areas where the code needs QA approval for merge

**Include code that touches the app:**
 
VAMobile/src 
VAMobile/e2e 
VAMobile/package.json 
VAMobile/.detoxrc.json

**Exclude PRs with only unit tests:**
VAMobile/src/*.test.tsx

**Resources**:

[Running a workflow when a pull request is approved](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved)

This will require attention from multiple engineers for testing since you can't approve your own PRs.

The idea that we're trying to codify here is that PRs shouldn't be merged into develop without QA approval.

**Comment by narin** [comment]
@timwright12 I was thinking @IsraelleHub could pick this one up soon. I saw that there is [an existing draft PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7562/files) that was closed. Is there anything salvageable from there? What's outstanding?

**Comment by timwright12** [comment]
@liztownd worked on that one, I think it was nearly there but got stale. I think it's probably a good starting point.

**Comment by bischoffa** [comment]
@IsraelleHub checking in on this ticket has its now been worked on in 3 sprints. Can you add a comment as to what you have done and what is causing this to extend over multiple sprints? Should this ticket be broken down into other tickets? Also can you size this ticket?

**Comment by IsraelleHub** [comment]
It was pending because I had to had more logic to it based on feedbacks. As of now, it should be good to go for final review of course before merging. If I need to add more logic to it, we can create another ticket for it. I did created separate ticket to make this workflow more efficient as well.

**Comment by IsraelleHub** [comment]
@bischoffa  Just FYII created a separate PR for this work. [https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8583](url) So most comment are there.

**Comment by bischoffa** [comment]
@IsraelleHub Checking on the status of this ticket. Last update was from 2 weeks or so ago. 

Is a PR still pending?

**Comment by bischoffa** [comment]
Also can we get a size added to this ticket?

**Comment by bischoffa** [comment]
@IsraelleHub can you post a message in engineering Slack that a PR is needed? May just need to remind engineers its open especially after the offsite.

**Comment by bischoffa** [comment]
Or if its a specific person can you reach out to them given its been pending for about 2 weeks.

**Comment by IsraelleHub** [comment]
@bischoffa Narin will be the one reviewing this one since I was working on some bug fix found while he was testing it.

**Comment by IsraelleHub** [comment]
@bischoffa  this is the current on going PR. [https://github.com/department-of-veterans-affairs/va-mobile-app/pull/8751](url)

**Comment by bischoffa** [comment]
Added existing sprint since this is still in PR

**Comment by bischoffa** [comment]
@narin @timwright12 @IsraelleHub what is the next steps with this PR?  I have tried to open the PR links listed above but they will not load. 

Per the comments PR was created a 27 days ago

**Comment by narin** [comment]
This is the actual PR: https://github.com/department-of-veterans-affairs/va-mobile-app/pull/8583

It just needs 1 QA review and it'll be good to go.

**Comment by bischoffa** [comment]
Thanks @narin! I see @rbontrager was tagged. Rachael if you are unable to we can chat with Therese to see if she can help.

**Comment by rbontrager** [comment]
@bischoffa Once this appts stuff is done I should have a little time to look at it. So I'm guessing early next week.

**Comment by bischoffa** [comment]
@IsraelleHub did someone in QA approve this yet? I know Rachael was asked but she won't get to it till next week due to H&B work so asked Tom to work with you on it and he posted last night he would get with you soon.

**Comment by IsraelleHub** [comment]
@bischoffa I missed out on those discussions. I will reopen it

**Comment by narin** [comment]
@bischoffa @IsraelleHub @rbontrager Historically, we haven't required QA to review/approve DevOps work. I think it's okay that this was merged and closed.

**Comment by bischoffa** [comment]
@narin was there a reason why you wanted QA to do it in your comment above?

**Comment by bischoffa** [comment]
I will support your decision to close it, I just thought you meant above a QA review being QA work. I likely misunderstood.

**Comment by bischoffa** [comment]
@narin @IsraelleHub can this ticket be closed? Haven't gotten a response to questions and also no one has closed the ticket.

**Comment by narin** [comment]
@bischoffa This was re-opened to address a bug with the original merge. The bug has since been fixed and merged a few days ago. Closing now.

**Comment by bischoffa** [comment]
Thanks! When reopening tickets please add a comment as to why its reopened so we can improve out documentation.
